### PR TITLE
Change NameValidator's type argument to an enum

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/GraphQL.ApiTests/GraphQL.approved.txt [hunk 1]
```diff
@@ -2507,11 +2507,19 @@ namespace GraphQL.Utilities
         public TType GetMetadata<TType>(string key, TType defaultValue = default) { }
         public bool HasMetadata(string key) { }
     }
+    public enum NameType
```
Comment (sungam3r) [inline src/GraphQL.ApiTests/GraphQL.approved.txt:2510]: Will there be confusion with `NamedType`?

Comment (Shane32) [reply src/GraphQL.ApiTests/GraphQL.approved.txt:2510]: I didn't think of that.  If you have a suggestion, I'm all ears.  I couldn't think of a good name.

Comment (Shane32) [reply src/GraphQL.ApiTests/GraphQL.approved.txt:2510]: If you would rather skip this PR, that's fine too, as it shouldn't really matter.  But in principle I think an enum is the right choice for the method.

File: src/GraphQL/Utilities/NameType.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+namespace GraphQL.Utilities
+{
+    /// <summary>
+    /// Indicates a field, type, argument, enum or directive.
+    /// </summary>
+    public enum NameType
```
Comment (sungam3r) [inline src/GraphQL/Utilities/NameType.cs:6]: NamedElement ? "Type" can be confusing.

File: src/GraphQL/Utilities/NameType.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        /// <summary>
+        /// An argument
+        /// </summary>
+        Argument,
+
+        /// <summary>
+        /// An enum
+        /// </summary>
+        Enum,
```
Comment (sungam3r) [inline src/GraphQL/Utilities/NameType.cs:26]: EnumValue ?

File: src/GraphQL.ApiTests/GraphQL.approved.txt (hunk 1 above)
Comment (sungam3r) [reply src/GraphQL.ApiTests/GraphQL.approved.txt:2510]: 50/50

Comment (joemcbride) [reply src/GraphQL.ApiTests/GraphQL.approved.txt:2510]: I like the use of the enum.  👍🏼

File: src/GraphQL/Utilities/NameType.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQL/Utilities/NameType.cs:6]: I like that

File: src/GraphQL/Utilities/NameType.cs (hunk 3 above)
Comment (Shane32) [reply src/GraphQL/Utilities/NameType.cs:26]: Ok

File: src/GraphQL/Utilities/NamedElement.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        /// </summary>
+        Type,
+
+        /// <summary>
+        /// An argument
+        /// </summary>
+        Argument,
+
+        /// <summary>
+        /// An enum
```
Comment (sungam3r) [inline src/GraphQL/Utilities/NamedElement.cs:24]:
```suggestion
        /// An enum value
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**